### PR TITLE
[Model] Support SenseNova U1 (NEOChatModel, MoT interleaved understanding+generation)

### PR DESCRIPTION
**Pull request by waltstephen**
## Summary

This PR adds native SGLang-Omni support for SenseNova U1
(`NEOChatModel`), including interleaved image understanding and generation.

The implementation includes:

- a native SGLang language/MoT model with direct checkpoint loading;
- native-resolution U1 vision encoding;
- U1 text/image hybrid attention masks in the real Triton serving path;
- native T2I/IT2I flow matching;
- native text-image-text interleaved generation;
- Radix prefix reuse, continuous batching, and CUDA graph acceleration;
- OpenAI-compatible inline image responses.

Official SenseNova U1 HF modeling is used only by validation scripts as an
oracle. Public serving factories fail fast if HF U1 modeling is imported.

## Why

SenseNova U1 is a unified multimodal model that can understand and generate
interleaved text and images. Downstream rollout workloads need a single native
engine instead of forwarding computation to Transformers or coordinating
separate autoregressive and image-generation services.

### Native model and weights

- Adds `SenseNovaU1NativeForCausalLM`.
- Uses SGLang `QKVParallelLinear`, `RowParallelLinear`,
  `VocabParallelEmbedding`, `RadixAttention`, and `LogitsProcessor`.
- Loads the U1 understanding and MoT generation towers directly from sharded
  safetensors.
- Implements split temporal/spatial RoPE with the checkpoint's independent
  text and image frequencies.
- Loads all 1,096 language-model keys with zero missing, unexpected, or error
  keys.

### Vision and hybrid attention

- Adds a native U1 vision model and native-resolution preprocessing.
- Implements the U1 row policy where text remains causal while image rows can
  attend to the complete image span.
- Passes custom masks through the actual SGLang Triton attention backend.
- Supports cached-prefix masks with a full current generated-image span.

### Flow matching and interleave

- Loads the U1 flow head, timestep/noise embedders, and generation vision tower
  directly from checkpoint weights.
- Reuses static condition-prefix KV through the Radix cache.
- Uses a breakable prefill CUDA graph for fixed generated-image token shapes.
- Implements native text boundary decode, flow image generation, official-order
  BF16 image re-embedding, and post-image continuation.
- Uses segmented exact eager CUDA graphs for interleaved text generation.

### Serving isolation

- Makes native interleave the public default variant.
- Keeps public `flow`, `interleave`, `native`, and `native_serving` variants on
  native implementations.
- Removes the HF VQA serving factory from public stages.
- Adds import guards to native flow/interleave/serving factories.

### Production safety bounds

- Uses a configurable 4-entry eager prefix LRU and 2-entry eager decode-graph
  LRU.
- Limits eager prefix entries to 2,048 tokens and decode graphs to 1,024 total
  tokens.
- Caps decode-graph capture at four lifetime attempts per executor. After the
  budget is exhausted, uncached shapes use the exact eager path while cached
  hot shapes continue to replay.
- Calls `CUDAGraph.reset()` and drops retained graph/tensor references on
  eviction; stage shutdown clears both eager caches.
- Guarantees prefill cleanup with `try/finally`, including sidecar, forward,
  logits-validation, metadata, and partial-batch failures.
- Validates image dimensions, pixels, diffusion steps, input/output image
  counts, output tokens, and the exact total-token budget before request GPU
  tensor allocation.
- Documents that tokenizer loading uses `trust_remote_code=True` and therefore
  requires a trusted, intentionally configured checkpoint.

## Validation

All results below were produced on one local H100 80GB. Each performance case
uses one unmeasured warmup/precompile pass followed by three measured repeats;
the median excludes model load.

### Cross-implementation correctness

- Fixed-input logits cosine: `0.9999924302`
- Logits max absolute difference: `0.0308227539`
- Greedy prompt matches: `20/20`
- Greedy token-step matches: `160/160`
- Native/HF image comparisons: `12/12`
- Minimum PSNR: `46.7483550 dB`
- Minimum SSIM: `0.99903745`
- Exact PNG count: `0`
- Ten-sample causal interleave correctness: `10/10`
- Same-image post-image continuation: `10/10` token exact

### Warmed performance

| Workload | HF | Native | Speedup |
|---|---:|---:|---:|
| Text bs=1 | 23.3347 tok/s | 58.4513 tok/s | 2.5049x |
| Text bs=8 | 23.2775 tok/s | 387.1545 tok/s | 16.6321x |
| Text bs=16 | 23.3750 tok/s | 729.8184 tok/s | 31.2222x |
| T2I, 256x256, 2 steps | 0.160354 s | 0.050109 s | 3.2001x |
| Interleave first-image fixed work | 6.296434 s | 2.350700 s | 2.6785x |

All text speed cases are token exact. T2I reaches PSNR `51.3745` and SSIM
`0.999650`; interleave reaches PSNR `51.0354` and SSIM `0.999623`.

The fixed-work interleave benchmark uses the shared exact 125-token pre-image
prefix, one image, and one post-image token so accepted pixel differences do
not change the measured amount of text work.

### Tests

```bash
python -m pytest \
  tests/unit_test/sensenova_u1/test_pipeline.py \
  tests/unit_test/sensenova_u1/test_native_model.py \
  tests/unit_test/sensenova_u1/test_hybrid_attention.py \
  tests/unit_test/sensenova_u1/test_safety.py -q
```

Result: `54 passed`.

The safety suite includes malformed/oversized payload rejection, LRU stress,
explicit graph reference release, lifetime capture budgeting, sidecar/forward/
logits/metadata fault injection, partial scheduling, and request/KV pool
baseline restoration.

Actual H100 cache stress across three cycles of varied prompt/decode lengths:

- prefix entries: `4/4`
- decode graph entries: `2/2`
- decode graph capture attempts: `4/4`
- cycle-end prefix retained bytes:
  `[122326016, 122326016, 122326016]`
- cycle-end graph retained bytes:
  `[598591696, 598591696, 598591696]`
- allocated growth after the warm cycle: `0` bytes
- reserved growth after the warm cycle: `4194304` bytes
- explicit clear returns both entry counts and retained bytes to zero

No-environment fixed-work spot after the safety changes:

- T2I: `3.1463x`, PSNR `51.3745`, SSIM `0.999650`
- interleave: `2.6776x`, PSNR `51.0354`, SSIM `0.999623`
- output PNG/text artifacts are byte-identical to the previously accepted
  no-environment run
- public HF import guard passes, including the native-serving factory

Review commits:

- `f40b0fd760b699a6006b0a55360f5f174dacf30b`
- `9fad08481aad342a19380eaa4a0f84c4f5f24db9`

## Known Limitations

- Validation is currently single-node and single-GPU.
- The production custom-mask path validated here is Triton; FlashInfer/FA3
  parity remains follow-up work.
- CUDA graph capture adds cold-start work that is excluded from warmed
  go/no-go measurements and retained separately in the evidence.
- CUDA graph-private allocator pools are not reliably reclaimed during process
  lifetime. The executor therefore limits new graph capture attempts to four;
  later uncached shapes use the exact eager path.
- Cooperative scheduling between individual flow steps and unrelated
  autoregressive requests remains future scheduler work.

### CODE REVIEW

**Comment by waltstephen** [comment]
Closing because this work targets the wrong repository. SenseNova U1 support will be re-targeted to the main sgl-project/sglang repository. Please do not merge this PR.
